fabric-tag-extensions-v0/src/main/java/net/fabricmc/fabric/api/tag/FabricTagBuilder.java

```diff
@@ -27,4 +29,16 @@
 	 * Clear the contained entries and mark the tag as replaced.
 	 */
 	void clearTagEntries();
+
+	/**
+	 * Add an optional object entry.
+	 * @param id The ID of the object to add
+	 */
+	void addOptional(Identifier id);
```

Comment:
`addOptionalObject` maybe to reduce confusion? Would also be nice to describe these in more detail. The reader should get an idea what these methods are needed/useful for.

Reply:
They correspond directly to the non-optional methods on the base vanilla class, so I didn't really want to go into a whole explanation of the tag system.

Reply:
There is no need to explain the whole tag system, just the differences Tag<>Object and Optional<>"Normal". E.g. something along the lines of "Optional entries will not produce load errors if their referenced object is missing" or whatever is appropriate.

Please consider adding this documentation, we try to guide the curious viewer to a reasonable extent. This file is particularly unusual/confusing since it's a detached interface that needs casting to be useful, I hope we can at some point migrate it to a subclass of Tag.Builder. Right now the two addOptional methods are quite exposed compared to the vanilla equivalents if someone sees our class before the vanilla one.

Reply:
done